### ISSUE TITLE
Data recovery (undeleting/salvaging files)

### ISSUE THREAD

**Issue by t-w** [t-w/ADFlib#9]
State and some notes what is still left to do.

- `adf_salv.c` module plus some tests in `regtests/` already existed
- `adf_salvage` command-line prototype was recently added

In general - after some corrections, simple cases (no lost/overwritten blocks) seem to work.

There are however special cases, that may require a lot more for proper data recovery:
- corrupted / overwritten file ext. blocks
  - can be circumvented on OFS (using metadata from data blocks)
- partial reconstruction if some data blocks were overwritten
- reconstruction of files without or with corrupted file header block (probably possible only on OFS, with part of metadata duplicated in data blocks)

(these might be done later)

-----------------

There are also few more error cases where it is not sure how to treat it:
- `adfUndelFile` in the last part of file recovery does:
```C
 if ( adfCreateEntry(vol, &parent, name, entry->headerKey) == -1 ) {
        rc = ADF_RC_ERROR;
        goto adfUndelFile_error_set_blocks_free;
    }
    if ( adfVolHasDIRCACHE ( vol ) ) {
        rc = adfAddInCache ( vol, &parent, (struct AdfEntryBlock *) entry );
        if ( rc != ADF_RC_OK )
            goto adfUndelFile_error_set_blocks_free;
    }
```
Basically, it is not quite clear what to do (or what is better to do) if `adfAddInCache` fails:
1. revert all changes, leaving filesystem unchanged
   - this requires reversing `adfCreateEntry` - it is not clear how to do it (`adfRemoveEntry`?)
2. if dir. cache is not critical and is updated later by the OS (this is to check/confirm), it might be reasonable not to fail in such case (maybe just issue a warning)
